payments-core/src/main/java/com/stripe/android/payments/core/injection/PaymentCommonModule.kt

```diff
@@ -69,28 +69,52 @@ internal class PaymentCommonModule {
 
     @Provides
     @Singleton
-    fun providePaymentFlowResultProcessor(
+    fun providePaymentIntentFlowResultProcessor(
+        appContext: Context,
+        lazyPaymentConfiguration: Lazy<PaymentConfiguration>,
+        stripeApiRepository: StripeApiRepository,
+        @Named(ENABLE_LOGGING) enableLogging: Boolean,
+    ): PaymentIntentFlowResultProcessor {
+        return PaymentIntentFlowResultProcessor(
+            appContext,
+            { lazyPaymentConfiguration.get().publishableKey },
+            stripeApiRepository,
+            enableLogging = enableLogging,
+            Dispatchers.IO
+        )
+    }
+
+    @Provides
+    @Singleton
+    fun provideSetupIntentFlowResultProcessor(
         appContext: Context,
-        clientSecret: ClientSecret,
         lazyPaymentConfiguration: Lazy<PaymentConfiguration>,
         stripeApiRepository: StripeApiRepository,
         @Named(ENABLE_LOGGING) enableLogging: Boolean,
+    ): SetupIntentFlowResultProcessor {
+        return SetupIntentFlowResultProcessor(
+            appContext,
+            { lazyPaymentConfiguration.get().publishableKey },
+            stripeApiRepository,
+            enableLogging = enableLogging,
+            Dispatchers.IO
+        )
+    }
+
+    /**
+     * Fetch the correct [PaymentFlowResultProcessor] based on current [ClientSecret].
+     *
+     * Should always be injected with [Provider].
+     */
+    @Provides
+    fun providePaymentFlowResultProcessor(
+        clientSecret: ClientSecret,
```

Comment:
Where does this value come from?

Reply:
As [documented](https://github.com/stripe/stripe-android/blob/92d767cbf43037d01655cfc03af15cf165ce22aa/payments-core/src/main/java/com/stripe/android/payments/core/injection/PaymentCommonModule.kt#L26), it should be provided by another `Module` when `PaymentCommonModule` is used, currently it is used in two places
1. [PaymentSheetViewModelComponent](https://github.com/stripe/stripe-android/blob/2e8a33981011660eebed6ce4070ffa6525bb1f02/payments-core/src/main/java/com/stripe/android/paymentsheet/injection/PaymentSheetViewModelComponent.kt#L15) - provided from [PaymentSheetViewModelModule](https://github.com/stripe/stripe-android/blob/2e8a33981011660eebed6ce4070ffa6525bb1f02/payments-core/src/main/java/com/stripe/android/paymentsheet/injection/PaymentSheetViewModelModule.kt#L43) through its starterArgs
2. [FlowControllerComponent](https://github.com/stripe/stripe-android/blob/2e8a33981011660eebed6ce4070ffa6525bb1f02/payments-core/src/main/java/com/stripe/android/paymentsheet/injection/FlowControllerComponent.kt#L21) - provided from [FlowControllerModule](https://github.com/stripe/stripe-android/blob/92d767cbf43037d01655cfc03af15cf165ce22aa/payments-core/src/main/java/com/stripe/android/paymentsheet/injection/FlowControllerModule.kt#L43) through `FlowControllerViewModel`

Reply:
Thanks for explaining. It might be helpful to link to that from here.